src/steamship/agents/schema/memory_window_strategy.py

```diff
@@ -0,0 +1,81 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import tiktoken
+from pydantic.main import BaseModel
+
+from steamship import Block
+from steamship.agents.memory.chathistory import ChatHistory
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class MemoryWindowStrategy(BaseModel, ABC):
```

Comment:
nit: i wonder if we should make the base: `MemoryStrategy` and have windowed stuff derive from that?